src/storage/FileResourceStore.ts

```diff
@@ -149,7 +149,7 @@ export class FileResourceStore implements ResourceStore {
     // Break up the request URI in the different parts `path` and `slug` as we know their semantics from addResource
     // to call the InteractionController in the same way.
     const [ , path, slug ] = /^(.*\/)([^/]+\/?)?$/u.exec(this.parseIdentifier(identifier)) ?? [];
-    if ((path === undefined || normalizePath(path) === '/') && slug === undefined) {
+    if ((typeof path === 'undefined' || normalizePath(path) === '/') && typeof slug === 'undefined') {
```

Comment:
Ah, I should have been clearer in my comment. I would prefer to state what we do expect instead of saying that something happens when it is undefined. So `typeof path !== 'string'` here, and similar for the others.